### ISSUE TITLE
Calculation of Account Health has gaps, display could be improved [Web]

### ISSUE THREAD

**Issue by laisee** [Power-Trade/api-docs#18]
On Web site I can enter an AVAX perpetual trade order. When I change the order quantity shows different values for Account Health after trade execution.

Enter order qty for 10, see health is "88%"

<img width="537" alt="Screenshot 2024-02-13 at 11 21 45 am" src="https://github.com/Power-Trade/api-docs/assets/5905130/f33a68a6-239f-4f60-a5ae-075c43e46002">

Enter  order qty for 100,  see health is "100%"

<img width="523" alt="Screenshot 2024-02-13 at 11 22 04 am" src="https://github.com/Power-Trade/api-docs/assets/5905130/d1cc832f-ea2e-4a5a-9b44-29e2cbf49ef0">

Enter  order qty for 80,  see health is "11%"

<img width="277" alt="Screenshot 2024-02-13 at 11 22 55 am" src="https://github.com/Power-Trade/api-docs/assets/5905130/9ffbdcc5-6825-4a55-9724-4beb8a21fe63">

Enter  order qty for 90,  see health is "0%"

<img width="294" alt="Screenshot 2024-02-13 at 11 23 24 am" src="https://github.com/Power-Trade/api-docs/assets/5905130/06eefe28-0b07-4570-8b98-f6bc52f19dbb">

Also, the margin balance is not being recalculated - depending on the order quantity  entered.

**Comment by klugjo** [comment]
> Enter order qty for 100, see health is "100%"

This is a "known" limitation of the system. If the order were to blow up your account, the system is incapable of calculating the margin available. cc @kernelbee 

> Also, the margin balance is not being recalculated - depending on the order quantity entered.

@laisee can you give me more details on what that means ? I am not too sure what that means, thanks

**Comment by laisee** [comment]
here are some screen shots taken from prod web UI

ADA Perpetual with order qty 10 - health @ 98%, margin is non-zero

<img width="245" alt="Screenshot 2024-02-14 at 3 33 01 pm" src="https://github.com/Power-Trade/api-docs/assets/5905130/d7020738-a40f-4658-a468-6c8ab78ee4f2">

ADA Perpetual with order qty 100 - health @ 97%, margin is non-zero

<img width="239" alt="Screenshot 2024-02-14 at 3 33 10 pm" src="https://github.com/Power-Trade/api-docs/assets/5905130/3273a689-46c7-447d-8fc7-4705abdd2822">

ADA Perpetual with order qty 1000  - health @ 83%, margin is non-zero

<img width="240" alt="Screenshot 2024-02-14 at 3 33 26 pm" src="https://github.com/Power-Trade/api-docs/assets/5905130/a91551ec-a024-41e8-b652-546328076d36">

ADA Perpetual with order qty 10000  - health now back @ 98% margin value at zero 

<img width="241" alt="Screenshot 2024-02-14 at 3 33 36 pm" src="https://github.com/Power-Trade/api-docs/assets/5905130/47eb6b32-afb1-4745-817d-1109709f90da">

reduce order quantity to 1000 and health / margin both recalculated to previous values

<img width="243" alt="Screenshot 2024-02-14 at 3 34 20 pm" src="https://github.com/Power-Trade/api-docs/assets/5905130/aee71307-39b4-437c-b6af-cbf57e1866e6">

**Comment by klugjo** [comment]
Thanks @laisee so no issue on the web app. 

Only issue here is that system limitation around calculating margin and account health when the result would liquidate the account.

This is a known bug

**Comment by laisee** [comment]
If the order size will cause liquidation should it not show new account
health as 0? As a trader I'd prefer to see visible clues and even a warning
that my trade is beyond account limits. Another option is to use a slider
with 25/50.75/100 percentage of available account collateral that can be
applied to a trade.

**Comment by klugjo** [comment]
I agree with you, unfortunately it is an API limitation that prevents us from doing that (instead of telling us that the account would blow up, it tells us nothing, making it undistinguishable from a "no margin usage for the trade" scenario). Hopefully a fix will be available soon

**Comment by laisee** [comment]
Understand - can change this issue to "Improvement in Account Health calculation & display".  Then leave it open till API allows improvement. How does that sound?

**Comment by klugjo** [comment]
sounds good thanks

**Comment by laisee** [comment]
Closing until the issue can be resolved with Vendor through API updates etc

**Comment by laisee** [comment]
re-opening as issue still seen in test and prod version of system.

**Comment by klugjo** [comment]
@laisee this has been improved:

In case there is not enough margin you get this message:

<img width="322" alt="image" src="https://github.com/Power-Trade/api-docs/assets/2960040/a31c3c72-4cfa-448b-ada7-70a7611a6c54">


We are still unable to compute exactly how much margin you would need when you don't have enough but at least you get notified that the order would not go through.

For cases where there are enough funds to cover margin, correct number is displayed:

<img width="321" alt="image" src="https://github.com/Power-Trade/api-docs/assets/2960040/7460f5e3-ea44-4fdc-9b8f-4269d061a35c">

**Comment by laisee** [comment]
Re-tested, found some improvements but still issues which can leave traders potentially losing funds without enough guidance. See example below where projected account health gaps after entering 346+ contracts and any number above that shows same "91%" projected health.

Issues:

1. Traders can't see their funds will be liquidated i.e. no warning
2. UI should not allow traders to pick new positions that will liquidate their funds - should be a slider allowing 25/50/75 % of available funds to be used, like many other exchanges. 

**Enter LONG AVAX Perp trade with 34 contracts -> new health is 82%**

<img width="356" alt="Screenshot 2024-07-22 at 17 25 54" src="https://github.com/user-attachments/assets/0c6ee102-02ce-413d-b22b-15818d94ad06">

**Enter LONG AVAX Perp trade with 300 contracts -> new health is 16%**

<img width="349" alt="Screenshot 2024-07-22 at 17 26 10" src="https://github.com/user-attachments/assets/b83d56f4-ad37-40d3-b676-e3f49806c8f7">

**Enter LONG AVAX Perp trade with 343 contracts -> new health is 05%**

<img width="347" alt="Screenshot 2024-07-22 at 17 26 21" src="https://github.com/user-attachments/assets/7464ed58-0151-46a6-9518-e3f3be76a704">

**Enter LONG AVAX Perp trade with 348 contracts -> new health is 91%**

<img width="345" alt="Screenshot 2024-07-22 at 17 26 32" src="https://github.com/user-attachments/assets/1b33e1f3-55f5-4986-8e33-5fe387e53854">

**Enter LONG AVAX Perp trade with 3488 contracts -> new health is 91%**

<img width="345" alt="Screenshot 2024-07-22 at 17 26 56" src="https://github.com/user-attachments/assets/b7d041ea-8cba-4f75-94a0-72aa3a9027b8">
